### DISCUSSION TITLE
Nextjs13 server component is running a fetch call twice on revalidate.

### DISCUSSION THREAD

**Discussion by rgioeli** [vercel/next.js#45551] [Help]
I have tried disabling reactStrictMode in next.config.js but it's not the problem. My code is just connecting to MongoDB and finding a user in the database and returning the user. Here is my component (I made it as simple as possible to test):

```
export const revalidate = 20;

async function getUsers() {
  const client = await clientPromise;
  const db = client.db("local");
  const data = await db
    .collection("users")
    .findOne({ email: "bob123@gmail.com" });
  console.log("THIS IS THE SERVER RUNNING");
  return data;
}

export default async function Home() {
  const user = await getUsers();
  if (!user) return notFound();
  return (
    <main className={styles.main}>
      {user && <p key={user._id}>{user.name}</p>}
    </main>
  );
}
```
Even in build mode, I still get my console.log() of "THIS IS THE SERVER RUNNING" printed out twice as you see below in the terminal. This happens every 20 seconds because of my revalidation setting.

![image](https://user-images.githubusercontent.com/45113824/216573620-3bcd6c1e-aef4-4b52-8fd2-67264b5a0cde.png)

My worry here is every 20 seconds I'm revalidating data twice, when it should only be once.

Anyone have any fix or explanation for this? Or could it possibly be a bug in Nextjs 13?

**Comment by rookbreezy** [comment]
I've got the same problem, on build it always runs twice, revalidate or not. It seems like something is wrong with `/app` static generation in general. Deploying to Vercel does not fix it.

Similar issues:
https://github.com/vercel/next.js/issues/44966
https://github.com/vercel/next.js/discussions/43158

**Reply by rgioeli** [reply]
So you too? Let me know if you run into anything that fixes it and I’ll do the same. For now I think I’ll head back to Next 12 because this could get expensive data wise.

**Reply by rookbreezy** [reply]
Yes, the `/app` folder is not ready for production yet. I've had a couple other major issues with it.

Next 13 should be fine to use. I'm just sticking with the `/pages` folder for a while longer.

**Reply by nabildroid** [reply]
same here, and a lot of another issues

**Comment by agamm** [comment]
Happens here too, problem I'm calling a function that uses `prisma.findMany` and it queries twice. Might be because I call it from the `metadata`  `generateMetadata` function too.

Any ETA or reference issue tackling this?

**Comment by palmmymumu** [comment]
Same here, `generateMetadata` was called twice when redirect from another page, but only once when direct page load.

**Comment by pranav-pandey** [comment]
Any updates here? I'm still encountering this with app router, the function on await gets triggered twice when it should be just once on the initial server component render

**Comment by GuysmoB** [comment]
Same here

**Reply by rookbreezy** [reply]
It's been 1.5 years since this issue was first talked about https://github.com/vercel/next.js/discussions/43158. Is it really still an issue? 

I have still not switched to the `/app` folder because of PTSD from all the bugs on from my previous attempts.

**Reply by shweta-codeprism** [reply]
yeah, its still an issue

**Reply by GuysmoB** [reply]
In my project I had a loading.tsx file next the main page and layout file.
After removing it, problem is gone. Why ? I have absolutely no idea but it's working. I guess I was using this file in the wrong way.
To find the solution,  I played  with the Vercel playground repository : https://app-router.vercel.app/
